src/quest.cpp

```diff
@@ -1023,45 +1064,38 @@ SPECIAL(johnson)
           forget(johnson, ch);
       }
       
-      // If you've done this quest recently, you can't do it again until you do more.
-      for (int i = QUEST_TIMER - 1; i >= 0; i--)
-        if (GET_LQUEST(ch, i) == quest_table[GET_SPARE2(johnson)].vnum) {
-          if (quest_table[GET_SPARE2(johnson)].done)
-            do_say(johnson, quest_table[GET_SPARE2(johnson)].done, 0, 0);
-          else {
-            snprintf(buf, sizeof(buf), "WARNING: Null string in quest %ld!", quest_table[GET_SPARE2(johnson)].vnum);
-            mudlog(buf, ch, LOG_SYSLOG, TRUE);
-            do_say(johnson, "I don't need help right now.", 0, 0);
-          }
-          if (memory(johnson, ch))
-            forget(johnson, ch);
-          return TRUE;
-        }
-      
       // Reject high-rep characters.
-      if (rep_too_high(ch, GET_SPARE2(johnson))) {
-        unsigned int johnson_max_rep = get_johnson_overall_max_rep(johnson);
-        if (johnson_max_rep < GET_REP(ch)) {
-          do_say(johnson, "My jobs aren't high-profile enough for someone with your rep!", 0, 0);
-          send_to_char(ch, "[OOC: This Johnson caps out at %d reputation, so you won't get any further work from them.]", johnson_max_rep);
-        } else {
-          do_say(johnson, "With rep as high as yours? I can't afford your rates for this one!", 0, 0);
-        }
+      unsigned int johnson_max_rep = get_johnson_overall_max_rep(johnson);
+      if (johnson_max_rep < GET_REP(ch)) {
+        do_say(johnson, "My jobs aren't high-profile enough for someone with your rep!", 0, 0);
```

Comment:
Good catch. This was the only untested scenario because my testers didn't have too high rep. I moved this check prior to the call to new_quest() and it should be working again.